### ISSUE TITLE
different policy inference time caused by delay inside env.step

### ISSUE THREAD

**Issue by a-z-e-r-i-l-a** [DLR-RM/stable-baselines3#86]
I have made a simple gym environment to check how fast the step method of this environment is called by the learner, and also how fast the inference time of the policy might be.
To do so I measure the time taken from the end of the last step call to the beginning of the current step as shown below and then print it. 

This duration is about 6 micorsecond from what I checked, however, when I put a delay of, for example 4 seconds inside the step function, this duration is increased up to 3ms which didn't make sense to me. Furthermore, without the delay also the step call on average seems to take 1 ms.

```python
class dummy_env(gym.Env):
    """Custom Environment that follows gym interface"""
    metadata = {'render.modes': ['dummy_env']}

    def __init__(self):
        super(dummy_env, self).__init__()

        self.observation_space = spaces.Box(low=-1, high=1, shape=(13,),  dtype=np.float32)
        self.action_space      = spaces.Box(low=-1, high=1, shape=(7,),   dtype=np.float32)

        self.action_inference_dur    = []
        self.episode_length          = 10000
        self.step_call_t                 = time.time()
        after_reset_or_train         = True


    def step(self, action):
        if not self.after_reset_or_train:
            self.action_inference_dur.append((time.time() - self.step_call_t))
            print(self.action_inference_dur[-1], 'on average:', np.mean(self.action_inference_dur))


        time.sleep(4)
        observation = np.array(13*[0])
        self.reward = 0
        if self.curr_episode_step == self.episode_length:
            self.done = True
        self.curr_episode_step += 1


        self.step_call_t = time.time()
        return observation, self.reward, self.done,  {}

    def reset(self):
        self.done=False
        self.curr_episode_step=0
        self.num_episodes += 1
        self.after_reset_or_train = True
        return np.array(13*[0])
```

**Comment by Miffyli** [comment]
With what algorithms this was tested? The "speed" of the environment should not affect the speed algorithms works at like noticed here (6µs -> 3ms), as in there is no connection with the two by design. My wild guess is that e.g. CPU/GPU clock-speeds are lowered because it does not sense a big load.

**Comment by PartiallyTyped** [comment]
> My wild guess is that e.g. CPU/GPU clock-speeds are lowered because it does not sense a big load.

Not as wild considering he is using sleep.

**Comment by a-z-e-r-i-l-a** [comment]
I had used PPO.

yea my thought was also that environment speed shouldn't effect the inference speen unless there is something because of global parameter lock in python going on, but with the environment above all the effect was caused by the sleep.

**Comment by araffin** [comment]
Hello,

I don't really get your problem, the way you are timing things is also missing.

**Comment by a-z-e-r-i-l-a** [comment]
yea, it would be more accurate to time the model.predict(), but I wanted to for example check how fast the implementation samples from the environment, that is how fast the step() is called, and for this I had measured the time distance from the end of the step until the next time when the step is called again.

So basically this is about increasing the sampling speed at train time. because in my real environment I had noticed that this duration was about 1 or 2 ms which wasn't clear why.
